src/detect-ftp-command.c

```diff
@@ -91,11 +93,8 @@ void DetectFtpCommandRegister(void)
     sigmatch_table[DETECT_FTP_COMMAND].Setup = DetectFtpCommandSetup;
     sigmatch_table[DETECT_FTP_COMMAND].flags |= SIGMATCH_NOOPT;
 
-    DetectAppLayerInspectEngineRegister(BUFFER_NAME, ALPROTO_FTP, SIG_FLAG_TOSERVER, 0,
-            DetectEngineInspectBufferGeneric, GetData);
-
-    DetectAppLayerMpmRegister(BUFFER_NAME, SIG_FLAG_TOSERVER, 2, PrefilterGenericMpmRegister,
-            GetData, ALPROTO_FTP, 1);
+    DetectHelperBufferMpmRegister(
```

Comment:
I don't think this is much of an improvement, if at all. We hide many details, register at a different state w/o explanation, the direction is now very unclear (false, true) means STREAM_TOSERVER apparently.

Reply:
Perhaps improving the helper registration functions and updating this logic is a path forward?

Reply:
Yeah we can do that post-beta1

Reply:
>  the direction is now very unclear (false, true) means STREAM_TOSERVER apparently.

Indeed we should change `DetectHelperBufferMpmRegister` to have a u8 instead of 2 bools